conf/pipeline/mag/snellius.config

```diff
@@ -0,0 +1,202 @@
+    withName: FILTLONG {
+        cpus   = { 32 * task.attempt }
+        memory = { 56.GB * (2 ** (task.attempt - 1)) }
+        time   = 5.d
+    }
+    withName: CENTRIFUGE {
+        cpus   = { 32 * task.attempt }
+        memory = { 56.GB * task.attempt }
+        time   = 5.d
+    }
+    withName: KRAKEN2 {
+        cpus   = { 32 * task.attempt }
+        memory = { 56.GB * task.attempt }
+        time   = 5.d
+    }
+    withName: KRONA {
+        cpus   = { 16 * task.attempt }
+        memory = { 28.GB * task.attempt }
+        time   = 5.d
+    }
+    withName: CAT_DB_GENERATE {
+        cpus =   { 32 * task.attempt }
+        memory = { 240.GB * task.attempt }
+        time   = 5.d
+        queue = 'fat_genoa'
+    }
+    withName: CAT {
+        cpus   = { 32 * task.attempt }
+        memory = { 56.GB * task.attempt }
+        time   = 5.d
+    }
+    withName: GTDBTK_CLASSIFYWF {
+        cpus   = { 32 * task.attempt }
+        memory = { 240.GB * task.attempt }
+        time   = 5.d
+        queue = 'fat_genoa'
+    }
+    //MEGAHIT returns exit code 250 when running out of memory
+    withName: MEGAHIT {
+        cpus          = { 32 * task.attempt }
+        memory        = { 56.GB * task.attempt }
+        time          = 5.d
+        errorStrategy = { task.exitStatus in ((130..145) + 104 + 250) ? 'retry' : 'finish' }
+    }
+    //SPAdes returns error(1) if it runs out of memory (and for other reasons as well...)!
+    //exponential increase of memory and time with attempts, keep number of threads to enable reproducibility
+    withName: SPADES {
+        cpus          = { 32 * task.attempt }
+        memory        = { 240.GB * (2 ** (task.attempt - 1)) }
+        time          = 5.d
+        errorStrategy = { task.exitStatus in [143, 137, 21, 1] ? 'retry' : 'finish' }
+        maxRetries    = 3
+        queue = 'fat_genoa'
+    }
+    withName: SPADESHYBRID {
+        cpus          = { 32 * task.attempt }
+        memory        = { 240.GB * (2 ** (task.attempt - 1)) }
+        time          = { 1.d * (2 ** (task.attempt - 1)) }
+        errorStrategy = { task.exitStatus in [143, 137, 21, 1] ? 'retry' : 'finish' }
+        maxRetries    = 3
+        queue = 'fat_genoa'
+    }
+    //returns exit code 247 when running out of memory
+    withName: BOWTIE2_ASSEMBLY_ALIGN {
+        cpus          = { 16 * task.attempt }
+        memory        = { 28.GB * task.attempt }
+        time          = { 24.h * (2 ** (task.attempt - 1)) }
+        errorStrategy = { task.exitStatus in [143, 137, 104, 134, 139, 247] ? 'retry' : 'finish' }
+    }
+    withName: METABAT2_METABAT2 {
+        cpus   = { 16 * task.attempt }
+        memory = { 28.GB * task.attempt }
+        time   = { 24.h * (2 ** (task.attempt - 1)) }
+    }
+    withName: MAG_DEPTHS {
+        cpus   = { 16 * task.attempt }
+        memory = { 28.GB * task.attempt }
+        time   = { 24.h * (2 ** (task.attempt - 1)) }
```

Comment:
I naturally don't know anything about your system, but the two main reasons to play with time is either fit in what's allowed for jobs to a specific partition or possibly to get squeezed in by the backfill (and in some very specific cases not being stopped because the job as submitted would be more than the total allocation).

Are you sure these give any benefit and just not make things needlessly complicated (with a not-insignificant risk of well functioning jobs timing out when they wouldn't need to)?